pkg/client/session.go

```diff
@@ -16,6 +16,13 @@ import (
 )
 
 func (c *immuClient) OpenSession(ctx context.Context, user []byte, pass []byte, database string) (err error) {
+	defer func() {
```

Comment:
Please add separate defer statements for each of the variables zeroed below. Those should be set up right after assigning value to them.

Also each of them will require some kind of deinitialization:

E.g. for the `SessionId` the defer right after assignment:

```go
	c.SessionID = resp.GetSessionID()
	defer func(){
		if err != nil {
			c.ServiceClient.CloseSession(ctx, new(empty.Empty))
			c.SessionID = nil
		}
	}()
```

For the `HeartBeater` it will require `c.HearBeater.Stop()`.

The `ServiceClient` member does not need deinitialization.

There's one more value to deinitialize in case of an error: `c.clientConn`, that one should be deinitialized with `c.clientConn.Close()` and set to nil too.